src/redisearch_rs/result_processor/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+pub mod counter;
+
+use std::{ffi::c_void, marker::PhantomPinned, mem::MaybeUninit, ptr::NonNull, time::Duration};
+
+#[derive(Default)]
+pub struct SearchResult {
+    // stub
+}
+
+pub enum Error {
+    // stub
+
+    // Result is empty, and the last result has already been returned.
+    //     EOF,
+    //     // Execution paused due to rate limiting (or manual pause from ext. thread??)
+    //     PAUSED,
+    //     // Execution halted because of timeout
+    //     TIMEDOUT,
+    //     // Aborted because of error. The QueryState (parent->status) should have
+    //     // more information.
+    //     ERROR,
+    //     // Not a return code per se, but a marker signifying the end of the 'public'
+    //     // return codes. Implementations can use this for extensions.
+    //     MAX,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+pub trait ResultProcessor {
+    fn next(&mut self) -> Result<Option<SearchResult>>;
```

Comment:
```suggestion
    fn next(&mut self, sr: &mut SearchResult) -> Result<Option<SearchResult>>;
```

To my best knowledge the SearchResult is owned by the caller of the pipeline.